### ISSUE TITLE
Pragma rules fail when query params present

### ISSUE THREAD

**Issue by rreusser** [threepointone/ratpack#28]
Is it possible that ratpack isn't handling query parameters in loader names correctly? I'm not all that familiar with webpack and am having some trouble setting up a loader. The obvious fix just passes the problem elsewhere so that I'm not quite sure how to fix it.

My `./ratpack/package.json` looks like this:

```json
{
  "name": "ratpack-local",
  "dependencies": {
    "brfs": "^1.4.3",
    "transform-loader": "^0.2.3"
  }
}
```

My `test.js` looks like this

```javascript
/* @ratpack {
  rules: [{
    files: '*.js',
    loader: "transform-loader?brfs",
    enforce: "post"
  }]
} */
```

I get the following error in the ratpack console:

<img width="506" alt="screen shot 2017-01-12 at 16 29 43" src="https://cloud.githubusercontent.com/assets/572717/21909063/598c539e-d8e4-11e6-9c54-33308bea64a7.png">

If I remove `?brfs` from the loader name, then the module lookup seems to succeed, except now I get this in the browser console:

<img width="861" alt="screen shot 2017-01-12 at 16 30 42" src="https://cloud.githubusercontent.com/assets/572717/21909102/7bebe9a4-d8e4-11e6-8af5-0658ade0a639.png">

I hate to put this on you (everything else is looking great!) but just wanted to check run by you the possibility that it might be a corner case in how ratpack is handling this. 🚀

**Comment by threepointone** [comment]
hrmm. ok lemme explain what's going on, for you, and posterity's sake. 

so, because of some quirks with electron archives and the lot, the loader isn't directly passed into the webpack config, but passed through `require.resolve()` to get exact paths before using. This means that the `?` syntax won't work, unfortunately. That said, I've actually moved away from using the archive format, so have some ideas for making this work. Will leave this issue open until then. 

Would you please tell me your usecase? I've included a bunch of loaders by default, so maybe you won't need it. Unless you're exploring a browserify compat codebase?

On a broader note, with the wealth of browserify transforms out there, I should consider supporting them natively. Will ponder :)

**Comment by rreusser** [comment]
No worries. Your project is great and I definitely don't want to pull it too heavily toward my peculiar use case at the expense of your time! I just saw a twitter survey indicating a strong general leaning toward webpack (selection bias aside). So I'd vote for keeping it in mind but otherwise neglecting it for now.

Yeah my use case is browserify-compat. My projects rely on [`glslify`](https://github.com/stackgl/glslify) and [`brfs`](https://github.com/substack/brfs) by way of plotly.js.

**Comment by threepointone** [comment]
you're speaking to a fellow browserify-fan here btw :) only seriously got into webpack sometime last year. There's tons of browserify code out there that I'd like to augment with ratpack, so your use case isn't that peculiar. 

quick sketch for a possible api 
```jsx
transforms: [ { files: '*.coffee', transform: 'coffeeify' ... } ]
```

**Comment by rreusser** [comment]
Hmm. Would you have to do:

```javascript
/* @ratpack {
  transforms: [ { files: '*.coffee', transform: 'coffeeify' ... } ]
} */
require('coffeeify');
```

In order to get it to install coffeeify as required by the transform, or is the transform name always identically equal to the module name? If not, feels like this might still depending on the order of operations, though if the file is preprocessed and node modules installed maybe that would work. [`detective`](https://github.com/substack/node-detective) is the browserify-style module that preprocesses and figures out the required modules. Not sure the webpack/babel equivalent though you must already be doing the equivalent. 😄

**Comment by threepointone** [comment]
more like 
```jsx
/* @ratpack {
  transforms: [ { files: '*.coffee', transform: 'coffeeify' ... } ]
} */
require('script.coffee');
```

**Comment by threepointone** [comment]
(you could just use a webpack coffeescript loader for this specific usecase, but consider brfs etc)

**Comment by rreusser** [comment]
Ah, I meant just to get the transform-containing module (e.g. `brfs`) installed in the first place.

**Comment by threepointone** [comment]
I'll share something in a day or so, I hope you'll be up for some beta testing :)

**Comment by rreusser** [comment]
Of course! Reading the source now so I can help out a bit instead of just making requests :)

**Comment by threepointone** [comment]
Lol the source is a mess, sorry in advance

**Comment by threepointone** [comment]
btw, for glsl, you could try https://www.npmjs.com/package/webpack-glsl-loader
```jsx
/* @ratpack {
  loaders: [ { files: '*.glsl', loader: 'webpack-glsl-loader' } ]
} */
let shader = require('./something.glsl')
// ...
```
and if you're *really* adventurous, you could try implementing a webpack loader that calls the brfs transform internally. (note to self - make sure `loader` works with relative paths).

**Comment by threepointone** [comment]
oh wait, glslify uses tagged literals, not the same as this loader. nevermind :|
